algorithms/src/polycommit/kzg10/data_structures.rs

```diff
@@ -206,6 +206,61 @@ pub struct VerifierKey<E: PairingEngine> {
     pub prepared_beta_h: <E::G2Affine as PairingCurve>::Prepared,
 }
 
+impl<E: PairingEngine> CanonicalSerialize for VerifierKey<E> {
+    fn serialize_with_mode<W: Write>(&self, mut writer: W, compress: Compress) -> Result<(), SerializationError> {
+        self.g.serialize_with_mode(&mut writer, compress)?;
+        self.gamma_g.serialize_with_mode(&mut writer, compress)?;
+        self.h.serialize_with_mode(&mut writer, compress)?;
+        self.beta_h.serialize_with_mode(&mut writer, compress)?;
+        Ok(())
+    }
+
+    fn serialized_size(&self, compress: Compress) -> usize {
+        self.g.serialized_size(compress)
+            + self.gamma_g.serialized_size(compress)
+            + self.h.serialized_size(compress)
+            + self.beta_h.serialized_size(compress)
+    }
+}
```

Comment:
Is it low hanging fruit to generate a proc macro for struct members at some point that would be similar to `#[serde(skip_serializing)]`. A proc macro  (something like `[canonical(skip_serializing)]`) that can apply to struct members would potentially allow the team to quickly create size savings like this.

Just a minor thought for the future.

Reply:
Yes, that would be a great contribution and time-saver!